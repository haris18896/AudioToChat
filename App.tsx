import React from 'react';

import { SafeAreaView, StyleSheet } from 'react-native';
import { ChatScreen } from './src/components';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <ChatScreen />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8FAFC',
  },
});

export default App;
