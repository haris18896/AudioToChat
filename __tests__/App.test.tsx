/**
 * @format
 */

import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import App from '../App';

// Mock the ChatScreen component
jest.mock('../src/components/ChatScreen', () => {
  const mockReact = require('react');
  return function MockChatScreen() {
    return mockReact.createElement('Text', {}, 'ChatScreen');
  };
});

// Mock SafeAreaProvider
jest.mock('react-native-safe-area-context', () => ({
  SafeAreaProvider: ({ children }: any) => children,
}));

describe('App', () => {
  it('renders correctly with test renderer', () => {
    expect(() => {
      const tree = ReactTestRenderer.create(<App />);
      tree.unmount();
    }).not.toThrow();
  });

  it('creates a valid component tree', () => {
    const tree = ReactTestRenderer.create(<App />);
    const instance = tree.getInstance();
    expect(instance).toBeDefined();
    tree.unmount();
  });

  it('matches snapshot structure', () => {
    const tree = ReactTestRenderer.create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });

  it('handles component lifecycle correctly', () => {
    const tree = ReactTestRenderer.create(<App />);
    expect(() => tree.unmount()).not.toThrow();
  });
});
