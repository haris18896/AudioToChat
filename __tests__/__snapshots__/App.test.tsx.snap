// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot structure 1`] = `null`;
